server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -382,16 +403,19 @@ static final class TopKLongState extends State {
         static final int ID = 2;
 
         private final LongsSketch sketch;
+        private final DataType<?> dataType;
         private final int limit;
 
-        TopKLongState(LongsSketch sketch, int limit) {
+        TopKLongState(LongsSketch sketch, DataType<?> dataType, int limit) {
```

Comment:
This is increasing the size of the state again (which can be per group in GROUP BY queries).

Any way to leave this out? Otherwise okay to continue with this until all the other optimizations are in and then maybe do another iteration over it.

Reply:
I know, I would prefer I do a clean up once the performance work is done.